core/dbt/adapters/base/impl.py

```diff
@@ -1298,6 +1298,44 @@ def render_column_constraint(cls, constraint: ColumnLevelConstraint) -> str:
         else:
             return ""
 
+    @classmethod
+    def _parse_model_constraint(cls, raw_constraint: Dict[str, Any]) -> ModelLevelConstraint:
+        try:
+            ModelLevelConstraint.validate(raw_constraint)
+            c = ModelLevelConstraint.from_dict(raw_constraint)
+            return c
+        except Exception:
+            raise DbtValidationError(f"Could not parse constraint: {raw_constraint}")
+
+    @available
+    @classmethod
+    def render_raw_model_constraints(cls, raw_constraints: Dict[str, Any]) -> str:
```

Comment:
Shouldn't the return type be a List here?
```suggestion
    def render_raw_model_constraints(cls, raw_constraints: Dict[str, Any]) -> List:
```